src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs

```diff
@@ -464,7 +472,7 @@ private const
                 power >>= 1;
             }
 
-            return result.Slice(0, resultLength);
```

Comment:
Was this slice basically unnecessary in all cases, and causing problems in this edge case?

Reply:
Yes, correct. It doesn't exist for overload of `PowCore` with exponent of arbitrary length.

Reply:
Removing this slice introduced a regression for the case of:
```
value: 174475926811173371108797680334274448482027679060635889276326576767485928691470272416521788770673419883076040463212449631887959313694345475337324466914914839368969988215395980439434099840700753576299320684517028939674493640310781015739837159625716992654178036326745036841619965661654305787656989089177185300627
exponent: 152266387011071825363527751452194927195391511201743678326712300944812006670853968954038119879530327815990523885596782666640476470053822766345969220504425389995206265094671158964311234812226260664473986637197447367726400093404731405598815953979171704175885394570388678952091206779039984338579147114424357147650
modulus: 15548277136038299103
```

Reply:
In this scenario, the length of `result` is greater than the length of `bits` and so the `result.CopyTo(bits)` operation fails as the destination is too short.

I expect the actual issue here is that `resultLength` isn't being correctly tracked somewhere.

Reply:
Its not that `resultLength` is incorrect, its that `bits` will have been mutated and may contain temporary data